src/app/handlers/command.handler.ts

```diff
@@ -114,42 +145,57 @@ export class CommandHandler extends Handler {
     return { name, args };
   }
 
-  private async _attemptRunPlugin(
-    message: types.IMessage,
-    plugin: types.IPlugin,
-    command: types.ICommand,
-    isDM: boolean
-  ) {
+  private async _attemptRunPlugin(commandData: CommandResolvable) {
+    const { isDM, plugin, message } = commandData;
+
     if ((isDM && !plugin.usableInDM) || (!isDM && !plugin.usableInGuild)) {
       return;
     }
 
     const permissionResponse = plugin.hasPermission(message);
     if (!isDM && permissionResponse !== true) {
-      message.reply(permissionResponse);
+      message.reply({ content: `${permissionResponse}` });
       return;
     }
 
-    if (!plugin.validate(message, command.args)) {
-      await message.reply(`Invalid arguments! Try: \`${Constants.Prefix}${plugin.usage}\``);
-      return;
+    if (commandData.type === 'message') {
+      if (!plugin.validate(commandData.message, commandData.command.args)) {
+        await message.reply(`Invalid arguments! Try: \`${Constants.Prefix}${plugin.usage}\``);
+        return;
+      }
     }
 
     if (!plugin.isActive) {
       await message.reply('This plugin has been deactivated.');
       return;
     }
 
+    const user = message instanceof CommandInteraction ? message.user.tag : message.author.tag;
+
     const pEvent: types.IPluginEvent = {
       status: 'starting',
       pluginName: plugin.name,
-      args: command.args,
-      user: message.author.tag,
+      args: [],
+      user,
     };
 
     try {
       this.container.loggerService.info(JSON.stringify(pEvent));
-      await plugin.execute(message, command.args);
+
+      // If it's a regular command, use the regular executor.
+      if (commandData.type === 'message') {
+        await plugin.execute(commandData.message, commandData.command.args);
+      }
+
+      // If it's a slash command, use the slash runner.
+
+      // This should never happen.
```

Comment:
I don't understand this comment. Why shouldn't it ever happen?

Reply:
Basically because of lines 186-188 only slash commands should fall through. So if we get something that doesn't look like a slash command, something is wrong. I agree this logic is laid out in a bit of a strange way.